app/scripts/views/base.js

```diff
@@ -362,18 +363,32 @@ define(function (require, exports, module) {
       }
     },
 
-    beforeRender: function () {
-      // Implement in subclasses. If returns false, or if returns a promise
-      // that resolves to false, then the view is not rendered.
-      // Useful if the view must immediately redirect to another view.
+    /**
+     * Called before rendering begins. If returns false, or if returns
+     * a promise that resolves to false, then the view is not
+     * rendered. Useful to immediately redirect to another view before
+     * rendering begins.
+     */
+    beforeRender () {
     },
 
-    afterRender: function () {
-      // Implement in subclasses
+    /**
+     * Called after the rendering occurs. Can be used to print an
+     * error message after the view is already rendered.
+     *
+     * @returns {Promise}
+     */
+    afterRender () {
+      // Override in subclasses
+      return p();
```

Comment:
I updated `afterRender` and `afterVisible` to always return promises to make chaining a bit simpler. I didn't do the same for `beforeRender` because subclass overrides often just return a boolean.